.github/workflows/mainPipeline.yml

```diff
@@ -0,0 +1,164 @@
+name: Main pipeline
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches: [ main ]
+  workflow_dispatch:
+    inputs:
+      override:
```

Comment:
The name `override` confused me a bit when I read it later in the file. Maybe `version` or `version_override` would be more clear?